design-proposals/sdn-ipam-secondary-nets.md

```diff
@@ -0,0 +1,490 @@
+the same IP address configured. This will surely lead to some dropped packets.
+
+Providing seamless live-migration is not one of the goals of this feature - a
+follow up epic will focus on that, by binding the logical switch port for the
+pod to the node where the VM is available in "real time" - i.e. initially the
+port is bound to the src node, then it becomes bound on both nodes, then
+finally it is bound exclusively to the dst node (when the destination pod takes
+over). This feature is called OVN multi-chassis binding; Openstack uses it
+successfully, as indicated in
+[this talk](https://www.youtube.com/watch?v=ijZTMXAg-eI).
+
+### KubeVirt related changes
+
+#### Pod templating
+
+KubeVirt would need to template the pod accordingly - i.e. the
+`NetworkSelectionElements` must feature the `ipam-claim-reference` attribute to
+indicate to the IPAM CNI / CNI which `IPAMClaim` they should use to persist the
+IP allocation across reboots / migrations.
+
+Check the following network selection elements annotation from a KubeVirt
+launcher pod to see an example:
+
+```bash
+kubectl get pods <launcher pod name> -ojsonpath="{ @.metadata.annotations.k8s\.v1\.cni\.cncf\.io\/networks }" | jq
+[
+  {
+    "name": "tenantred",
+    "namespace": "default",
+    "interface": "pod16367aacb67",
+    "ipam-claim-reference": "vm-a.tenantred"
+  }
+]
+```
+
+#### Creating the IPAMClaim
+KubeVirt must create the IPAMClaims (and set finalizers / owner references) to
+properly ensure the VM's lifecycle is reflected in the allocation.
+
+There are two important sections to fill when creating the IPAMClaim reference:
+- the IPAMClaim name
+- the spec.network attribute
+
+The `IPAMClaim` name will be created using the following pattern:
+`<vm name>.<logical network name>`. The logical network name is the [name of the
+network](https://kubevirt.io/api-reference/main/definitions.html#_v1_network)
+in the KubeVirt API.
+
+The `IPAMClaim.spec.network` must be the name of network as per the
+[CNI spec](https://github.com/containernetworking/cni/blob/main/SPEC.md#configuration-format).
+This attribute is mandatory as per the CNI specification, and the CNI requires
+it to understand how to map IPAMClaims to the networks.
+
+Below you can find an example of a network attachment definition, specifying
+the behavior of attachments to the `tenantblue` network, along with an
+`IPAMCLaim` for that network.
+```yaml
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: tenantblue
+spec:
+  config: |2
+    {
+            "cniVersion": "0.4.0",
+            "name": "tenantblue",
+            "type": "ovn-k8s-cni-overlay",
+            "topology":"localnet",
+            "subnets": "192.168.10.0/24",
+            "excludeSubnets": "192.168.10.1/32",
+            "allowPersistentIPs": true,
+            "netAttachDefName": "default/tenantblue"
+    }
+---
+apiVersion: k8s.cni.cncf.io/v1alpha1
+kind: IPAMClaim
+metadata:
+  name: vm1.tenantblue
```

Comment:
The fact the kubevirt logical network name and the OVN network name are both `tenantblue` is confusing. Please use different names (and add the VM/VMI snippet).